components/forwarder/forwarder_test.go

```diff
@@ -0,0 +1,176 @@
+package forwarder_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/components/forwarder"
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/ThreeDotsLabs/watermill/pubsub/gochannel"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	logger = watermill.NewStdLogger(true, true)
+
+	forwarderTopic = "forwarder_topic"
+	outTopic       = "out_topic"
+)
+
+// TestForwarder tests forwarding messages from PubSubIn to PubSubOut (which are GoChannel implementation underneath).
+func TestForwarder(t *testing.T) {
+	// Create a set of publisher and subscribers for both In and Out Pub/Subs.
+	publisherIn, subscriberIn := newPubSubIn(logger)
```

Comment:
maybe it would be better to create forwarder per every test? to be sure that we don't have any side effects (we can also run them in parallel in that case)

Reply:
Good idea: f72ba48